src/main/router.ts

```diff
@@ -103,78 +86,101 @@ export class Router {
       raw: {
         req,
       },
-      response: res,
       query: url.searchParams,
       payload,
       params,
     };
   }
+}
 
-  protected static async handleStaticFile({ params, response }: LensApiRequest): Promise<void> {
-    let filePath = params.path;
-
-    for (let retryCount = 0; retryCount < 5; retryCount += 1) {
-      const asset = path.join(Router.rootPath, filePath);
-      const normalizedFilePath = path.resolve(asset);
+export interface LensApiResult<TResult> {
+  statusCode?: number;
+  response?: TResult;
+  error?: any;
+  contentType?: LensApiResultContentType;
+  headers?: { [name: string]: string };
+}
 
-      if (!normalizedFilePath.startsWith(Router.rootPath)) {
-        response.statusCode = 404;
+export type RouteHandler<TResponse> = (
+  request: LensApiRequest
+) =>
+  | Promise<LensApiResult<TResponse>>
+  | Promise<void>
+  | LensApiResult<TResponse>
+  | void;
 
-        return response.end();
-      }
+export interface Route<TResponse> {
+  path: string;
+  method: "get" | "post" | "put" | "patch" | "delete";
+  handler: RouteHandler<TResponse>;
+}
 
-      try {
-        const data = await readFile(asset);
+const handleRoute = (route: Route<unknown>) => async (request: LensApiRequest, response: http.ServerResponse) => {
+  let result: LensApiResult<any> | void;
 
-        response.setHeader("Content-Type", getMimeType(asset));
-        response.write(data);
-        response.end();
-      } catch (err) {
-        if (retryCount > 5) {
-          logger.error("handleStaticFile:", err.toString());
-          response.statusCode = 404;
+  const writeServerResponse = writeServerResponseFor(response);
 
-          return response.end();
-        }
+  try {
+    result = await route.handler(request);
+  } catch(error) {
+    const mappedResult = contentTypes.txt.resultMapper({
+      statusCode: 422,
```

Comment:
Should this be `500`?

Reply:
> Should this be `500`?

Previously these were 422. But yeah, seems wrong. Changed.